src/main/java/net/neoforged/neoforge/event/entity/player/ServerPlayerEvent.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.event.entity.player;
+
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.level.GameType;
+import net.minecraft.world.level.Level;
+import net.neoforged.bus.api.ICancellableEvent;
+import net.neoforged.neoforge.event.entity.living.LivingEvent;
+
+public abstract class ServerPlayerEvent extends LivingEvent {
```

Comment:
Would extending `PlayerEvent` not make more sense, giving that `ServerPlayer` extends from `Player`?

Reply:
I was thinking the same thing but I un-did it for some reason I can't remember